src/services/CurrencyService.ts

```diff
@@ -43,39 +41,30 @@ export function useFetchHistoryData(selectedCurrencies: CurrencyHistoryData[], p
     const [data, setData] = useState<HistoryData[]>([])
     const [loading, setLoading] = useState<boolean>(true)
 
-    const request = (url: string) => from(fetch(url)
-        .then(data => data.json()));
-
-    const checkIfPolishCurrencySelected = selectedCurrencies.findIndex(currency =>  currency.code === polishCurrencyCode)
+    const checkIfPolishCurrencySelected = selectedCurrencies.findIndex(currency => currency.code === polishCurrencyCode)
 
     useEffect(() => {
-
-        console.log('useEffect')
-
-        if (checkIfPolishCurrencySelected) {
-            selectedCurrencies = selectedCurrencies.filter(selectedCurrency => {
-                return selectedCurrency.code !== polishCurrencyCode
-            })
-        }
-
-        console.log(selectedCurrencies)
-
-        if (selectedCurrencies.length === 2) {
+        if (checkIfPolishCurrencySelected === -1) {
             forkJoin({
                 firstCurrency: request(`http://api.nbp.pl/api/exchangerates/rates/${selectedCurrencies[0]?.table}/${selectedCurrencies[0]?.code}/last/${period}`),
                 secondCurrency: request(`http://api.nbp.pl/api/exchangerates/rates/${selectedCurrencies[1]?.table}/${selectedCurrencies[1]?.code}/last/${period}`)
             }).subscribe(({firstCurrency, secondCurrency}) => {
-                setData(prepareHistoryData(firstCurrency.rates, secondCurrency.rates));
+                setData(prepareHistoryData(firstCurrency.rates, false, secondCurrency.rates));
                 setLoading(false)
 
             })
         }
 
-        if (selectedCurrencies.length === 1) {
+        if (checkIfPolishCurrencySelected > -1) {
+
+            selectedCurrencies = selectedCurrencies.filter(selectedCurrency => {
+                return selectedCurrency.code !== polishCurrencyCode
+            })
+
             forkJoin({
                 currency: request(`http://api.nbp.pl/api/exchangerates/rates/${selectedCurrencies[0]?.table}/${selectedCurrencies[0]?.code}/last/${period}`)
```

Comment:
What do you think about fetching data only once, on initialization, and not every time when someone change the `period` value. Then only display corresponding data based on active `period` value? Can we make this happen? 
This way we will display data faster and we significantly reduce amount of API calls. 

We can discuss about this f2f.

Reply:
Done as requested little Radzio